nntrainer/src/tensor.cpp

```diff
@@ -673,14 +685,37 @@ void Tensor::read(std::ifstream &file) {
 /**
  * @brief Calculate average value according to the axis.
  */
-Tensor Tensor::average(int axis) const {
+Tensor Tensor::average(unsigned int axis) const {
+  if (axis >= MAXDIM)
+    throw std::out_of_range(
+      "negative axis or axis more then MAXDIM is invalid");
+
   unsigned int axis_size = dim.getDim()[axis];
   if (axis_size == 1)
     return this->clone();
 
   return this->sum(axis, 1.0 / ((float)axis_size));
 }
 
+Tensor Tensor::average(const std::vector<unsigned int> &axes) const {
+  if (axes.size() > MAXDIM)
```

Comment:
Maybe this is redundant when line 709 check exists.

Reply:
FYI, line 701 checks if given axes should be less than 4. for example {0, 1, 2, 3, 0} is invalid. 
I'll allow the same axes passing twice with deleting these bits.

line 709 checks if each element is over maxdim for example {0, 3, 5} is invalid.

Reply:
Right, however {0,1,2,0} is still allowed. 
https://github.com/nnstreamer/nntrainer/pull/513#discussion_r485320976 can address this.
But should {0,1,2,0} be treated any different from {0,1,2,2,0}, I am not so sure.

So, even if axis.sizes() is more than MAXDIM, it will be covered in the checks later.
But when the input is correct, this is just a redundant check.

Reply:
Rearranged. PTAL

Reply:
Looks good now.